### PR TITLE
Refactor of Harmony base sample project

### PR DESCRIPTION
**Pull request by vilanovi**
**Note**: This PR includes https://github.com/mobilejazz/harmony-swift/pull/23. First, accept https://github.com/mobilejazz/harmony-swift/pull/23 and then this one.

- Removed Swinject and built a custom ApplicationProvider injection
- Fixed mockable.io API to work
- Moved all "core" code to a new folder structure
- Removed unused realm examples
- Refactored ItemList view controller to common MVP structure

### CODE REVIEW

**Comment by vilanovi** [comment]
Reviewers, please check the ApplicationProvider.swift and ItemProvider.swift to ensure everything is ok and following the standards we have. It's the first time I've written manually the DI, and I could have done wrongly many things (like, for example, the Mappers are being created within the ItemDefaultModule instead by constructor..?)

File: Example/Harmony/Core/Features/Item/Domain/Presenters/ItemDetailPresenter.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+//
+
+import Foundation
+
+protocol ItemDetailPresenterView: class {
+    func onDisplay(_ item: Item)
+}
+
+protocol ItemDetailPresenter {
+    func onSelectedItem(_ item: Item)
```
Comment (joselufo) [inline Example/Harmony/Core/Features/Item/Domain/Presenters/ItemDetailPresenter.swift]: These two functions are not using the prefixes specified in the harmony documentation. 

https://harmony.mobilejazz.com/docs/best-practices/presenter-specs

File: Example/Harmony/Core/Features/Item/Domain/Presenters/ItemListPresenter.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    func onDisplay(items: [Item])
+    func onNavigateTo(item: Item)
+    func onDisplayFailedToFetchItems(_ error: Error)
+}
+
+protocol ItemListPresenter {
+    func onEventLoadList()
+    func onActionSelected(item : Item)
+    func onActionReloadList()
+}
```
Comment (joselufo) [inline Example/Harmony/Core/Features/Item/Domain/Presenters/ItemListPresenter.swift]: Not using the prefixes defined in harmony. 
https://harmony.mobilejazz.com/docs/best-practices/presenter-specs

**Review by joselufo — CHANGES_REQUESTED** [review]

**Review by jfilowk — APPROVED** [review]

**Comment by vilanovi** [comment]
@joselufo changes applied. Please verify.

**Review by joselufo — APPROVED** [review]

**Pull request merged**